src/utils/format_graphite/format_graphite.c

```diff
@@ -34,6 +34,36 @@
 /* Utils functions to format data sets in graphite format.
  * Largely taken from write_graphite.c as it remains the same formatting */
 
+/* helper function for reverse_hostname */
+void reverse_string(char *r_host, int len) {
+  char t;
+  for (int i = 0, j = len - 1; i < j; i++, j--) {
+    t = r_host[i];
```

Comment:
Hi,

Thanks for your work on Collectd!

Just a minor note: in this place, `char t` had to be declared inside of the loop.

Reply:
> Just a minor note: in this place, `char t` had to be declared inside of the loop.

I was under the impression that keeping the declaration outside of the loop would be a tiny bit more efficient. I did some reading and learned that this is not the case. In fact, looking at the disassembled code with objdump showed no differences between the two versions. (I didn't use objdump before, so this was an interesting thing to do.) Many thanks for pointing it out to me!

Reply:
Compilers are very smart now.
Did you ever try to develop Atmel ATtiny13 microcontroller programs?
It has 1Kb program flash memory only, so ... .

I has one project with tiny13. When I looked into `objdump` output I found compiler optimizations which I ever can't think about. I think, If I will write assembler code, it will not be more than 1-3% smaller (by size) than gcc generated. But I will spent much-much more time to write and support such code.

Yes, that is an interesting experience. )